catalog/dags/common/utils.py

```diff
@@ -0,0 +1,69 @@
+import functools
+from typing import Any
+
+from common.constants import SQL_INFO_BY_MEDIA_TYPE
+
+
+def setup_kwargs_for_media_type(
+    values_by_media_type: dict[str, Any], kwarg_name: str
+) -> callable:
+    """
+    Create a decorator which provides media_type-specific information as parameters
+    for the called function. The called function must itself have a media_type kwarg,
+    which is used to select values.
```

Comment:
Not sure if it's worth it, but thought I'd mention that `inspect.signature` can list the parameters for a function, including explicitly defined kwargs. If desirable, a configuration check could be added in `wrap` that confirms the function is correctly configured for the decorator (including that `kwarg_name` is present). This wouldn't cover `**kwargs` use cases though.

Reply:
This is great! I struggled a bit with this, because `media_type` doesn't just need to be a keyword argument in the signature -- it needs to be passed to the called function as a keyword argument, never as a positional one. Calling this out has made me realize I should update it to be a [keyword-only argument](https://peps.python.org/pep-3102/#:~:text=Keyword%2Donly%20arguments%20are%20not,therefore%20'required%20keyword'%20arguments.), and it looks like I should be able to add a configuration check for that explicitly :)